test/unit/assorted/server_discovery_and_monitoring.spec.test.ts

```diff
@@ -464,8 +472,12 @@ function assertTopologyDescriptionOutcomeExpectations(
       if (WIRE_VERSION_KEYS.has(expectedKey) && expectedValue === null) {
         // For wireVersion keys we default to zero instead of null
         expect(actualServer).to.have.property(expectedKey, 0);
-      } else {
+      } else if (expectedKey !== 'error') {
         expect(actualServer).to.have.deep.property(expectedKey, expectedValue);
+      } else {
+        // pull the message off the error if it is present
+        expect(actualServer).to.have.deep.property(expectedKey).instanceof(MongoError);
```

Comment:
```suggestion
        expect(actualServer).to.have.property(expectedKey).instanceof(MongoError);
```

From the docs, `deep.property` makes all property assertions use ===, which isn't relevant here